lib/parachain/available_data_fetching.go

```diff
@@ -0,0 +1,109 @@
+package parachain
+
+import (
+	"fmt"
+
+	"github.com/ChainSafe/gossamer/lib/common"
+	"github.com/ChainSafe/gossamer/pkg/scale"
+)
+
+// AvailableDataFetchingRequest represents a request to retrieve all available data for a specific candidate.
+type AvailableDataFetchingRequest struct {
+	// Hash of the candidate for which the available data is requested.
+	CandidateHash CandidateHash
+}
+
+// Encode returns the SCALE encoding of the AvailableDataFetchingRequest
+func (a *AvailableDataFetchingRequest) Encode() ([]byte, error) {
+	return scale.Marshal(*a)
+}
+
+// AvailableDataFetchingResponse represents the possible responses to an available data fetching request.
+type AvailableDataFetchingResponse scale.VaryingDataType
+
+// NewAvailableDataFetchingResponse returns a new available data fetching response varying data type
+func NewAvailableDataFetchingResponse() AvailableDataFetchingResponse {
+	vdt := scale.MustNewVaryingDataType(AvailableData{}, NoSuchData{})
+	return AvailableDataFetchingResponse(vdt)
+}
+
+// Set will set a value using the underlying  varying data type
+func (a *AvailableDataFetchingResponse) Set(val scale.VaryingDataTypeValue) (err error) {
+	vdt := scale.VaryingDataType(*a)
+	err = vdt.Set(val)
+	if err != nil {
+		return
+	}
+	*a = AvailableDataFetchingResponse(vdt)
+	return
+}
+
+// Value returns the value from the underlying varying data type
+func (a *AvailableDataFetchingResponse) Value() (val scale.VaryingDataTypeValue, err error) {
+	vdt := scale.VaryingDataType(*a)
+	return vdt.Value()
+}
+
+// AvailableData represents the data that is kept available for each candidate included in the relay chain.
+type AvailableData struct {
+	// The Proof-of-Validation (PoV) of the candidate
+	PoV PoV `scale:"1"`
```

Comment:
This `PoV` type is `[]byte`

But the type for `PoV` here is 
```
pub struct PoV {
	/// The block witness data.
	pub block_data: BlockData,
}
```
BlockData being the following
```
pub struct BlockData(#[cfg_attr(feature = "std", serde(with = "bytes"))] pub Vec<u8>);
```